tests/Microsoft.Identity.Test.Unit/BrokerTests/WamAadPluginTests.cs

```diff
@@ -0,0 +1,354 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using NSubstitute;
+using Windows.Security.Authentication.Web.Core;
+using Windows.Security.Credentials;
+
+namespace Microsoft.Identity.Test.Unit.BrokerTests
+{
+    [TestClass]
+    public class WamAadPluginTests : TestBase
+    {
+        private ICoreLogger _logger;
+        private IWamPlugin _aadPlugin;
+        private IWamProxy _wamProxy;
+        private IWebAccountProviderFactory _webAccountProviderFactory;
+        private IAccountPickerFactory _accountPickerFactory;
+        private ICacheSessionManager _cacheSessionManager;
+        private IInstanceDiscoveryManager _instanceDiscoveryManager;
+
+        private MsalTokenResponse _msalTokenResponse = new MsalTokenResponse
+        {
+            IdToken = MockHelpers.CreateIdToken(TestConstants.UniqueId, TestConstants.DisplayableId),
+            AccessToken = "access-token",
+            ClientInfo = MockHelpers.CreateClientInfo(),
+            ExpiresIn = 3599,
+            CorrelationId = "correlation-id",
+            RefreshToken = null, // brokers don't return RT
+            Scope = TestConstants.s_scope.AsSingleString(),
+            TokenType = "Bearer",
+            WamAccountId = "wam_account_id",
+        };
+
+        [TestInitialize]
+        public void Init()
+        {
+            _logger = Substitute.For<ICoreLogger>();
+            _wamProxy = Substitute.For<IWamProxy>();
+            _webAccountProviderFactory = Substitute.For<IWebAccountProviderFactory>();
+            _accountPickerFactory = Substitute.For<IAccountPickerFactory>();
+
+            _webAccountProviderFactory.ClearReceivedCalls();
+            _cacheSessionManager = Substitute.For<ICacheSessionManager>();
+
+            _instanceDiscoveryManager = Substitute.For<IInstanceDiscoveryManager>();
+
+            _aadPlugin = new AadPlugin(_wamProxy, _webAccountProviderFactory, _logger);
+        }
+
+        [TestMethod]
+        public async Task GetAccounts_NoAccounts_Async()
+        {
+            // Arrange
+            using (MockHttpAndServiceBundle harness = CreateTestHarness())
+            {
+
+                // Act
+                var accounts = await _aadPlugin.GetAccountsAsync(
+                    TestConstants.ClientId,
+                    TestConstants.AuthorityCommonTenant,
+                    _cacheSessionManager,
+                    _instanceDiscoveryManager).ConfigureAwait(false);
+
+                // Assert
+                Assert.AreEqual(0, accounts.Count());
+            }
+        }
+
+        [TestMethod]
+        public async Task GetAccounts_WamAccounts_NoAuthority_Async()
+        {
+            // Arrange
+            using (MockHttpAndServiceBundle harness = CreateTestHarness())
+            {
+                var wamAccountProvider = new WebAccountProvider("id", "user1@contoso.com", null);
+                _webAccountProviderFactory.GetAccountProviderAsync("organizations").Returns(wamAccountProvider);
```

Comment:
would we ever create the webaccountProviderFactory w/out the authority or tenant?

Reply:
There's a special string you can pass in to get the web account provider of the "OS default account" - `https://login.windows.local`

This corresponds to the user from the "Your Info" settings page in Windows

![image](https://user-images.githubusercontent.com/12273384/106518125-5744c800-64d1-11eb-92b9-70383c84db8b.png)


https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/cb1361f5c235fdf893957ea145e299cf4686f0df/src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/WebAccountProviderFactory.cs#L24

Reply:
that's cool. 

i guess i'm asking more is this test an actual scenario, where we wouldn't pass the authority. just curious.